src/realizations/catchment/Bmi_Multi_Formulation.cpp

```diff
@@ -303,6 +303,23 @@ string Bmi_Multi_Formulation::get_output_line_for_timestep(int timestep, std::st
         // This almost certainly should never happen, but just to be safe ...
         if (output_var_names.empty()) { return ""; }
 
+        std::cout << "Multi: timestep = " << timestep << std::endl;
+        for (int i = 0; i < output_var_names.size(); ++i) {
+            std::cout << "Multi: output_var_names," << output_var_names[i] << std::endl;
+        }
+
+        // Remove invalid output variable name entry in the globle bmi-multi module
+        std::vector<std::string>::iterator it;
+        int last_index = modules.size() - 1;
+        auto module = modules[last_index];
```

Comment:
Actually this is wrong. The list of valid variable names for a multi_bmi formulation is _not_ found just in the last BMI module in the list; rather it is the union of all names of all BMI modules, including the forcing DataProvider. (Using the outputs from the last module is only relevant as the _default_ set of names used when nothing is explicitly specified, as done further down in this function.)

Perhaps this is why the test below was disabled? Does it attempt to retrieve a variable that is not from the last BMI module in the multi_bmi?

Reply:
It is possible we may get an incorrect header in the output if we check all the modules.

Reply:
How? _Checking_ all of the modules to see if a given variable is valid does not cause all of the modules' outputs to appear in the output--what causes them to appear in the output is if the user has explicitly called for them using `output_variables`.